src/Schema.php

```diff
@@ -210,15 +211,31 @@ protected function loadTableForeignKeys(string $tableName): array
         DbArrayHelper::multisort($foreignKeysList, 'seq');
 
         /** @psalm-var NormalizePragmaForeignKeyList $foreignKeysList */
-        foreach ($foreignKeysList as $table => $foreignKey) {
-            $fk = (new ForeignKeyConstraint())
-                ->columnNames(DbArrayHelper::getColumn($foreignKey, 'from'))
-                ->foreignTableName($table)
-                ->foreignColumnNames(DbArrayHelper::getColumn($foreignKey, 'to'))
-                ->onDelete($foreignKey[0]['on_delete'] ?? null)
-                ->onUpdate($foreignKey[0]['on_update'] ?? null);
-
-            $result[] = $fk;
+        foreach ($foreignKeysList as $table => $foreignKeys) {
+            $foreignKeysById = DbArrayHelper::index($foreignKeys, null, ['id']);
+
+            /** @psalm-var NormalizePragmaForeignKeyList $foreignKeysById */
+            foreach ($foreignKeysById as $id => $foreignKey) {
+                if ($foreignKey[0]['to'] === null) {
+                    $primaryKey = $this->getTablePrimaryKey($table);
+
+                    if ($primaryKey !== null) {
+                        /** @psalm-var string $primaryKeyName */
+                        foreach ((array) $primaryKey->getColumnNames() as $i => $primaryKeyName) {
+                            $foreignKey[$i]['to'] = $primaryKeyName;
+                        }
+                    }
+                }
+
+                $fk = (new ForeignKeyConstraint())
+                    ->columnNames(array_column($foreignKey, 'from'))
+                    ->foreignTableName($table)
+                    ->foreignColumnNames(array_column($foreignKey, 'to'))
+                    ->onDelete($foreignKey[0]['on_delete'])
+                    ->onUpdate($foreignKey[0]['on_update']);
+
+                $result[(int) $id] = $fk;
```

Comment:
```suggestion
                $result[] = $fk;
```

`loadTableForeignKeys` returns non-indexed array of foreign keys, right?

Reply:
Right, looks like need to store `$id` in `ForeignKeyConstraint::$name`